luigi_pipeline/seqr_loading_optimized.py

```diff
@@ -40,33 +39,45 @@ def run(self):
         if self.subset_path:
             mt = self.subset_samples_and_variants(mt, self.subset_path)
 
-        mt = SeqrGenotypesSchema(mt).annotate_all(overwrite=True).select_annotated_mt()
+        mt = self.GenotypesSchema(mt).annotate_all(overwrite=True).select_annotated_mt()
 
         mt.describe()
         mt.write(self.output().path, stage_locally=True, overwrite=True)
 
 
-class SeqrMTToESOptimizedTask(HailElasticSearchTask):
+class SeqrVCFToGenotypesMTTask(BaseVCFToGenotypesMTTask):
+    VariantTask = SeqrVCFToVariantMTTask
+    GenotypesSchema = SeqrGenotypesSchema
+
+
+class BaseMTToESOptimizedTask(HailElasticSearchTask):
 
     def __init__(self, *args, **kwargs):
         # TODO: instead of hardcoded index, generate from project_guid, etc.
         super().__init__(*args, **kwargs)
 
     def requires(self):
-        return [SeqrVCFToVariantMTTask(), SeqrVCFToGenotypesMTTask()]
+        return [self.VariantTask(), self.GenotypesTask()]
 
     def run(self):
         variants_mt = hl.read_matrix_table(self.input()[0].path)
         genotypes_mt = hl.read_matrix_table(self.input()[1].path)
+        genotypes_mt = genotypes_mt.drop(*[k for k in genotypes_mt.globals.keys()])
```

Comment:
Good catch! The following statement `row_ht = genotypes_mt.rows().join(variants_mt.rows())` joins two tables with the same global fields. Hail renames the global fields from one table to avoid conflicts and puts them all into the new table. Hail has the below messages:
```
2022-07-12 10:04:57 Hail: INFO: Table.join: renamed the following fields on the right to avoid name conflicts:
    'sourceFilePath' -> 'sourceFilePath_1'
    'hail_version' -> 'hail_version_1'
    'datasetType' -> 'datasetType_1'
    'sampleType' -> 'sampleType_1'
    'genomeVersion' -> 'genomeVersion_1'
```
Then, there are duplications in the global fields. All these fields go to the `_meta` field of the ES index while exporting the ES. It seems to me that we never use the fields "*_1" from the ES index. But I'm not sure whether they are useful or not.